src/main/java/org/sonarqube/auth/bitbucket/BitbucketIdentityProvider.java

```diff
@@ -103,6 +110,22 @@ public void callback(CallbackContext context) {
 
     GsonUser gsonUser = requestUser(scribe, accessToken);
     GsonEmails gsonEmails = requestEmails(scribe, accessToken);
+
+    String[] teamsRestriction = settings.teamRestriction();
```

Comment:
Please extract all this code in a specific method (something like `checkTeamRestriction(scribe, accessToken)`), to increase readability.